datafusion/physical-plan/src/stream.rs

```diff
@@ -408,7 +422,18 @@ where
     type Item = Result<RecordBatch>;
 
     fn poll_next(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<Self::Item>> {
-        self.project().stream.poll_next(cx)
+        let this = self.project();
+        let transform_schema = *this.transform_schema;
+        let schema = Arc::clone(this.schema);
+        let ret = this.stream.poll_next(cx);
+        if transform_schema {
+            if let Poll::Ready(Some(Ok(batch))) = ret {
+                return Poll::Ready(Some(batch.with_schema(schema).map_err(|e| {
```

Comment:
I think this is one of the notorious problems when logical schema doesn't match the physical one on nullability/metadata. But this change might bring a performance impact, although the schema change is just reassigning the value but it also calls `schema_contains` which may be expensive